test/distributed/org/apache/cassandra/distributed/impl/IsolatedJmx.java

```diff
@@ -84,14 +87,17 @@ public void startJmx() {
             String hostname = addr.getHostAddress();
             wrapper = new MBeanWrapper.InstanceMBeanWrapper(hostname + ":" + jmxPort);
             ((MBeanWrapper.DelegatingMbeanWrapper) MBeanWrapper.instance).setDelegate(wrapper);
-            Map<String, Object> env = new HashMap<>();
 
-            serverSocketFactory = new CollectingRMIServerSocketFactoryImpl(addr);
-            env.put(RMIConnectorServer.RMI_SERVER_SOCKET_FACTORY_ATTRIBUTE,
-                    serverSocketFactory);
-            clientSocketFactory = new RMIClientSocketFactoryImpl(addr);
-            env.put(RMIConnectorServer.RMI_CLIENT_SOCKET_FACTORY_ATTRIBUTE,
-                    clientSocketFactory);
+            // CASSANDRA-18508: Sensitive JMX SSL configuration options can be easily exposed
+            Map<String,Object> encryptionOptionsMap = (Map<String, Object>) config.getParams().get("jmx_encryption_options");
+            EncryptionOptions jmxEncryptionOptions = getJmxEncryptionOptions(encryptionOptionsMap);
+            // Here the `localOnly` is always passed as true as it is for the local isolated JMX testing
+            // However if the `jmxEncryptionOptions` are provided or JMX SSL configuration is set it will configure
+            // the socket factories appropriately.
+            Map<String, Object> socketFactories = new IsolatedJmxSocketFactory().configure(addr, true, jmxEncryptionOptions);
+            serverSocketFactory = (RMICloseableServerSocketFactory) socketFactories.get(RMIConnectorServer.RMI_SERVER_SOCKET_FACTORY_ATTRIBUTE);
```

Comment:
@maulin-vasavada do we still need to cast?

Reply:
yes because the return map has `Object` type so we have to cast.